kernel/init.c

```diff
@@ -363,13 +366,15 @@ static void prepare_multithreading(struct k_thread *dummy_thread)
 	}
 
 	initialize_timeouts();
+
+	return ret;
 }
 
-static FUNC_NORETURN void switch_to_main_thread(void)
+static FUNC_NORETURN void switch_to_main_thread(size_t stack_size)
 {
 #ifdef CONFIG_ARCH_HAS_CUSTOM_SWAP_TO_MAIN
 	arch_switch_to_main_thread(&z_main_thread, z_main_stack,
-				   K_THREAD_STACK_SIZEOF(z_main_stack),
+				   stack_size,
```

Comment:
I think K_THREAD_STACK_SIZEOF was here to not allow passing the whole z_main_stack size, and remove the guard if needed.